src/utils/streams.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const fs = require('fs');
+const fsPromises = require('fs/promises');
+const program = require('commander');
+const csvjson = require('csvjson');
+const axios = require('axios');
+
+// example url with css file for css bundler
+const url = 'https://gist.githubusercontent.com/eirikbakke/1059266/raw/d81dba46c76169c2b253de0baed790677883c221/gistfile1.css';
+
+// action functions
+const reverse = (str) => {
+  console.log(str.split('').reverse().join(''));
+};
+
+const transform = (str) => {
+  console.log(str.toUpperCase());
+};
+
+const outputFile = (filePath) => {
+  const read = fs.createReadStream(filePath, 'utf8');
+  read.on('error', err => console.log(`reading error: ${err}`));
+  read.pipe(process.stdout);
+};
+
+const convertFromFile = (filePath) => {
+  const read = fs.createReadStream(filePath, 'utf8');
+  read.on('error', err => console.log(`reading error: ${err}`));
+  const toObject = csvjson.stream.toObject();
+  const stringify = csvjson.stream.stringify();
+  read
+    .pipe(toObject)
+    .pipe(stringify)
+    .pipe(process.stdout);
+};
+
+const convertToFile = (filePath) => {
+  const read = fs.createReadStream(filePath, 'utf8');
```

Comment:
I can suggest separation of logic from IO. Split all your functions on 2, one of them will just create IO objects such as read stream or write stream, and call functions with logic, which will process data, but without working with IO directly. Such approach recommended both for OOP (separation of concerns, single responsibility in SOLID) and for FP (pure functions, they shouldn't produce side effects such as working with IO). Your code will became for simple, more readable and more composable - for example you will create convert method which will accept not filePath, but readStream and writeStream, in such case it will be extremely easy to create 2 different wrappers, one of them will read file and provide stream to convert, and second wrapper will provide just stdin/stdout, so your script will be able to use as a pipeline for bash scripts `cat file.csv | streams.js > out.json`. yes, it will take some effort to make your script executable, but the main point not about that - right now your functions are monolithic, they are responsible for too many things, so it's hard to reuse them anywhere else. The most reusable code is not complicated one, the most reusable code is the simple one, and best form of simplification is splitting by responsibility.